etc/scripts/freeze_and_update_reqs.py

```diff
@@ -0,0 +1,134 @@
+# http://nexb.com and https://github.com/nexB/scancode-toolkit/
+# The ScanCode software is licensed under the Apache License version 2.0.
+# Data generated with ScanCode require an acknowledgment.
+# ScanCode is a trademark of nexB Inc.
+#
+# You may not use this software except in compliance with the License.
+# You may obtain a copy of the License at: http://apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software distributed
+# under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+# CONDITIONS OF ANY KIND, either express or implied. See the License for the
+# specific language governing permissions and limitations under the License.
+#
+# When you publish or redistribute any data created with ScanCode or any ScanCode
+# derivative work, you must accompany this data with the following acknowledgment:
+#
+#  Generated with ScanCode and provided on an "AS IS" BASIS, WITHOUT WARRANTIES
+#  OR CONDITIONS OF ANY KIND, either express or implied. No content created from
+#  ScanCode should be considered or used as legal advice. Consult an Attorney
+#  for any legal advice.
+#  ScanCode is a free software code scanning tool from nexB Inc. and others.
+#  Visit https://github.com/nexB/scancode-toolkit/ for support and download.
+
+from __future__ import absolute_import
+from __future__ import print_function
+
+import argparse
+import fnmatch
+from commoncode.fileutils import resource_iter
+import os
+from subprocess import run
+from shutil import copy, rmtree
+import sys
+
+python_version = str(sys.version_info[0]) + str(sys.version_info[1])
+py_abi = "{0}cp{1}{0}".format("*", python_version)
+
+
+def generate_req_text(input_dir, req_file, package_name, upgrade):
+    """
+    Generate a requirement file at `output_file`(by default requirements.txt) of all dependencies wheels and sdists present in the `input_dir` directory.
+    If a `package_name` is provided it will be updated to its latest version.
+    """
+    thirdparty = resource_iter(input_dir, with_dirs=False)
+    dependencies = [
+        files
+        for files in thirdparty
+        if fnmatch.fnmatchcase(files, '*py3*')
+        or fnmatch.fnmatchcase(files, py_abi)
+        or (
+            fnmatch.fnmatchcase(files, '*tar.gz*')
+            and not fnmatch.fnmatchcase(files, '*py2-ipaddress-3.4.1.tar.gz*')
+        )
+    ]
+    if not (os.path.isdir('temp dir')):
+        os.mkdir('temp dir')
+    for deps in dependencies:
+        copy(deps, 'temp dir')
+    pip_args = [
+        'pip-compile',
+        '--generate-hashes',
+        '--find-links',
+        'temp dir',
+        '--output-file',
+        req_file,
+        '--allow-unsafe',
+        '--pip-args',
+        '--no-index',
+    ]
+    if upgrade:
+        pip_args.append('--upgrade')
+    if package_name:
+        pip_args.extend(['--upgrade-package', package_name])
+    run(pip_args)
+    rmtree('temp dir')
```

Comment:
With a proper https://docs.python.org/3/library/tempfile.html?highlight=tempfile#module-tempfile you will NOT need to rm anything, with will be handled for you if you use a context manager.
such as this example from the Python doc:
```
# create a temporary directory using the context manager
>>> with tempfile.TemporaryDirectory() as tmpdirname:
...     print('created temporary directory', tmpdirname)
 .... do your things here
```